### PR TITLE
[graphics]: add rewards graphics

### PR DESCRIPTION
**Pull request by dfkadyr**

### CODE REVIEW

**Review by mike-diamond — APPROVED** [review]

File: src/methods/vault/requests/getUserRewards/index.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      dateFrom,
+    } as UserRewardsQueryVariables,
+    modifyResult: modifyUserRewards,
+  })
+
+  return data
+}
+
+
+export default getUserRewards
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/methods/vault/requests/getUserRewards/index.ts:35]: https://github.com/stakewise/v3-sdk/pull/25/commits/67fe7f204c33aaed9b775037b5c192dccabebe7a#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R178

File: src/methods/vault/requests/getUserRewards/index.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { UserRewardsQueryVariables } from '../../../../graphql/backend/vault'
+import { apiUrls, validateArgs } from '../../../../utils'
+import modifyUserRewards from './modifyUserRewards'
+import { backend } from '../../../../graphql'
+import { ModifyUserReward } from './types'
+
+
+type GetUserRewardsInput = {
+  options: StakeWise.Options
+  address: UserRewardsQueryVariables['user']
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/methods/vault/requests/getUserRewards/index.ts]: https://github.com/stakewise/v3-sdk/pull/25/commits/67fe7f204c33aaed9b775037b5c192dccabebe7a#diff-c403463e96bc45d37c98445e5d14cd59550daddb013b22da36a7ef69bd8f1621R10

File: src/methods/vault/requests/getUserRewards/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  vaultAddress: UserRewardsQueryVariables['vaultAddress']
+  dateFrom: UserRewardsQueryVariables['dateFrom']
+}
+
+const getUserRewards = async (input: GetUserRewardsInput) => {
+  const { options, vaultAddress, address, dateFrom } = input
+
+  validateArgs.address({ address })
+  validateArgs.address({ vaultAddress })
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/methods/vault/requests/getUserRewards/index.ts]: https://github.com/stakewise/v3-sdk/pull/25/commits/67fe7f204c33aaed9b775037b5c192dccabebe7a#diff-c403463e96bc45d37c98445e5d14cd59550daddb013b22da36a7ef69bd8f1621R18

File: src/methods/vault/requests/getUserRewards/modifyUserRewards.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    acc[Number(date)] = modifyUserReward(rest)
+
+    return acc
+  }, {} as ModifyUserReward['days'])
+
+  return { days }
+}
+
+
+export default modifyUserRewards
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/methods/vault/requests/getUserRewards/modifyUserRewards.ts:26]: https://github.com/stakewise/v3-sdk/pull/25/commits/67fe7f204c33aaed9b775037b5c192dccabebe7a#diff-2879502b648fde23c7eeb91b3dd7204834d973b82652315405245543dde4d0b0R1

**Pull request merged**